tools/benchmarks/classification/knn_imagenet/test_knn.py

```diff
@@ -0,0 +1,184 @@
+
+    cfg = mmcv.Config.fromfile(args.config)
+    if args.cfg_options is not None:
+        cfg.merge_from_dict(args.cfg_options)
+    # set cudnn_benchmark
+    if cfg.get('cudnn_benchmark', False):
+        torch.backends.cudnn.benchmark = True
+    # work_dir is determined in this priority: CLI > segment in file > filename
+    if args.work_dir is not None:
+        # update configs according to CLI args if args.work_dir is not None
+        cfg.work_dir = args.work_dir
+    elif cfg.get('work_dir', None) is None:
+        # use config filename as default work_dir if cfg.work_dir is None
+        work_type = args.config.split('/')[1]
+        cfg.work_dir = osp.join('./work_dirs', work_type,
+                                osp.splitext(osp.basename(args.config))[0])
+
+    # init distributed env first, since logger depends on the dist info.
+    if args.launcher == 'none':
+        distributed = False
+    else:
+        distributed = True
+        init_dist(args.launcher, **cfg.dist_params)
+
+    # create work_dir and init the logger before other steps
+    timestamp = time.strftime('%Y%m%d_%H%M%S', time.localtime())
+    knn_work_dir = osp.join(cfg.work_dir, 'knn/')
+    mmcv.mkdir_or_exist(osp.abspath(knn_work_dir))
+    log_file = osp.join(knn_work_dir, f'knn_{timestamp}.log')
+    logger = get_root_logger(log_file=log_file, log_level=cfg.log_level)
+
+    # build the dataloader
+    dataset_cfg = mmcv.Config.fromfile(args.dataset_config)
+    dataset_train = build_dataset(dataset_cfg.data.train)
+    dataset_val = build_dataset(dataset_cfg.data.val)
+    if 'imgs_per_gpu' in cfg.data:
+        logger.warning('"imgs_per_gpu" is deprecated. '
+                       'Please use "samples_per_gpu" instead')
+        if 'samples_per_gpu' in cfg.data:
+            logger.warning(
+                f'Got "imgs_per_gpu"={cfg.data.imgs_per_gpu} and '
+                f'"samples_per_gpu"={cfg.data.samples_per_gpu}, "imgs_per_gpu"'
+                f'={cfg.data.imgs_per_gpu} is used in this experiments')
+        else:
+            logger.warning(
+                'Automatically set "samples_per_gpu"="imgs_per_gpu"='
+                f'{cfg.data.imgs_per_gpu} in this experiments')
+        cfg.data.samples_per_gpu = cfg.data.imgs_per_gpu
+    data_loader_train = build_dataloader(
+        dataset_train,
+        samples_per_gpu=dataset_cfg.data.samples_per_gpu,
+        workers_per_gpu=dataset_cfg.data.workers_per_gpu,
+        dist=distributed,
+        shuffle=False)
+    data_loader_val = build_dataloader(
+        dataset_val,
+        samples_per_gpu=dataset_cfg.data.samples_per_gpu,
+        workers_per_gpu=dataset_cfg.data.workers_per_gpu,
+        dist=distributed,
+        shuffle=False)
+
+    # build the model
+    model = build_algorithm(cfg.model)
+    model.init_weights()
+
+    # model is determined in this priority: init_cfg > checkpoint > random
+    if getattr(cfg.model.backbone.init_cfg, 'type', None) == 'Pretrained':
```

Comment:
Add one line to check the `init_cfg`
Refer to: https://github.com/open-mmlab/mmselfsup/pull/256